posts/loggingPython.md

```diff
@@ -0,0 +1,291 @@
+def captureException():
+	# this should do something
+
+try:
+	1/0
+except:
+	captureException()
+```
+
+When capturing an exception, it’s essential to add context. Similar to when you’re debugging, you want to know who the user is and what they were doing when the exception struck.
+
+```python
+import sys
+
+# remember to set up your logger
+
+def captureException():
+	logger.warning(sys.exc_info())
+```
+
+![](./images/loggingPython/exception1.png)
+
+This gives us a `traceback` object that we can use to see the stack from our exception. We didn’t even need to pass anything into our `captureException` method. 
+
+Let’s figure out what the `traceback` object gives us and clean up this information a little.
+
+```python
+import sys
+import traceback
+
+def captureException():
+	r = list(sys.exc_info())
+	
+	e = dict()
+	e["name"] = r[1]
+```
+
+![](./images/loggingPython/exception2.png)
+
+Now, this can be cleaned up using some string manipulation in Python.
+
+## Accessing those Logs
+
+Now that you’ve created _gorgeous_ logs that give the next developer a view of the user’s actions, you need to know best practices for searching and reacting to these logs.
+
+### Searching
+
+Going through your logs from a file is a tedious process, it’s honestly better to use a [cloud-based solution](https://timber.io/) that acts as a layer of abstraction, so you don’t have to think about parsing colossal log files.
+
+If you’re dead-set on searching through your files locally, Python is one of the best languages to do so. Here is an example that generalizes the text and log file:
+```python
+def search(input_filename, output_filename, text):
+	# overwrite output file
+	with open(output_filename, "w") as out_file:
+		with open(input_filename) as in_file:
+			# loop over each log line and check if text appears
+			for line in in_file:
+				if text in line:
+					out_file.write(line)
+```
+
+This makes it easy to search for log messages that contain keywords such as `critical` or `warning`.
+
+### Rotating Logs
+
+Though not generally taught in logging guides, the Python logging module makes it easy to log in a different file after an interval of time or after the log file reaches a certain size.
+
+This becomes useful if you automatically want to get rid of older logs, or if you're going to search through your logs by date, since you won’t have to search through a huge file to find a set of logs that are already grouped.
+
+To create a `handler` that makes a new log file every day and automatically deletes logs more than five days old, you can use a `TimedRotatingFileHandler`. Here’s an example:
+
+```python
+logger = logging.getLogger('Rotating Log by Day')
+
+# writes to pathToLog
+# creates a new file every day because `when="d"` and `interval=1`
+# automatically deletes logs more than 5 days old because `backupCount=5`
+handler = TimedRotatingFileHandler(pathToLog, when="d", interval=1,  backupCount=5)
+```
+
+### Accessing Server Logs through SSH
```

Comment:
I think this could be better, any ideas how?

Reply:
I would omit this section entirely.